src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/set_global_properties_request.cc

```diff
@@ -820,6 +831,112 @@ bool SetGlobalPropertiesRequest::IsWhiteSpaceExist() {
         }
       }
     }
+
+    if (msg_params[strings::keyboard_properties].keyExists(
+            hmi_request::customize_keys)) {
+      const smart_objects::SmartArray* custom_keys_array =
+          msg_params[strings::keyboard_properties][hmi_request::customize_keys]
+              .asArray();
+
+      smart_objects::SmartArray::const_iterator it_keys =
+          custom_keys_array->begin();
+      smart_objects::SmartArray::const_iterator it_keys_end =
+          custom_keys_array->end();
+
+      for (; it_keys != it_keys_end; ++it_keys) {
+        str = (*it_keys).asCharArray();
+        if (!CheckSyntax(str)) {
+          SDL_LOG_ERROR(
+              "Invalid keyboard_properties "
+              "customize_keys syntax check failed");
+          return true;
+        }
+      }
+    }
+  }
+
+  return false;
+}
+
+hmi_apis::Common_KeyboardLayout::eType
+SetGlobalPropertiesRequest::GetKeyboardLayout() {
+  SDL_LOG_AUTO_TRACE();
+
+  const smart_objects::SmartObject& msg_params =
+      (*message_)[strings::msg_params];
+  if (msg_params[strings::keyboard_properties].keyExists(
+          hmi_request::keyboard_layout)) {
+    return static_cast<hmi_apis::Common_KeyboardLayout::eType>(
+        msg_params[strings::keyboard_properties][hmi_request::keyboard_layout]
+            .asInt());
+  }
+
+  ApplicationSharedPtr app = application_manager_.application(connection_key());
+  auto saved_keyboard_props = app->keyboard_props();
+  if (saved_keyboard_props) {
+    if (saved_keyboard_props->keyExists(hmi_request::keyboard_layout)) {
+      return static_cast<hmi_apis::Common_KeyboardLayout::eType>(
+          (*saved_keyboard_props)[hmi_request::keyboard_layout].asInt());
+    }
+  }
+
+  return hmi_apis::Common_KeyboardLayout::QWERTY;
+}
+
+uint32_t SetGlobalPropertiesRequest::GetAllowedNumberOfConfigurableKeys() {
+  SDL_LOG_AUTO_TRACE();
+
+  auto display_capabilities = hmi_capabilities_.system_display_capabilities();
+  if (display_capabilities) {
```

Comment:
@ychernysheva I recommend to use reverse conditions to avoid so big nested-if-spagetty:
```
if (!display_capabilities) {
 SDL_LOG_ERROR("Capabilities are not available")'
 return 0;
}

if (!window_capabilities) {
...
}

if (!keyExists(keyboard_capabilities)) {
...
}

for...
```

It might look longer but much easier to read

Reply:
Done in f10bc21